### PR TITLE
feat(kratos): add statefulset.extraArgs option

### PR DESCRIPTION
**Pull request by Davincible**
Also fix capitalization in readme.md

Fix: #486

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: helm/charts/kratos/README.md [hunk 1]
```diff
@@ -100,15 +100,16 @@ A ORY Kratos Helm chart for Kubernetes
...
-| statefulSet.annotations | object | `{}` |  |
-| statefulSet.extraContainers | object | `{}` | If you want to add extra sidecar containers. |
-| statefulSet.extraEnv | list | `[]` |  |
-| statefulSet.extraInitContainers | object | `{}` | If you want to add extra init containers. These are processed before the migration init container. |
-| statefulSet.extraVolumeMounts | list | `[]` |  |
-| statefulSet.extraVolumes | list | `[]` | If you want to mount external volume For example, mount a secret containing Certificate root CA to verify database TLS connection. |
-| statefulSet.labels | object | `{}` |  |
-| statefulSet.nodeSelector | object | `{}` | Node labels for pod assignment. |
-| statefulSet.resources | object | `{}` |  |
+| statefulset.annotations | object | `{}` |  |
```
Comment (Demonsthere) [inline helm/charts/kratos/README.md:103]: The docs are auto-generated, no need to modify it :)

File: helm/charts/kratos/templates/statefulset-mail.yaml [hunk 2]
```diff
@@ -54,7 +54,14 @@ spec:
         - name: {{ include "kratos.fullname" . }}-courier
           image: {{ include "kratos.image" . }}
           imagePullPolicy: {{ include "kratos.imagePullPolicy" . }}
-          args: [ "courier", "watch", "--config", "/etc/config/kratos.yaml" ]
+          args:
+            - courier
+            - watch 
+            - --config
+            - /etc/config/kratos.yaml
+            {{- if .Values.statefulset.extraArgs }}
```
Comment (Demonsthere) [inline helm/charts/kratos/templates/statefulset-mail.yaml:62]: Please add the new parameter to the values.yaml with a default value of `[]`.

A good idea would be also to add some testing parameters to the values in the `/hack` directory, to ensure that the manifests are properly generated :)

**Review by Demonsthere — CHANGES_REQUESTED** [review]

File: helm/charts/kratos/README.md (hunk 1 above)
Comment (Davincible) [reply helm/charts/kratos/README.md:103]: I see. I think my changes can just be overwritten with the generated docs right?

**Comment by Davincible** [comment]
I was confused as in the same helm template, both `statefulSet` and `statefulset` were used. I changed all to the former since it aligns more with the rest of the conventions used throughout the template. This is a breaking change, however.

File: helm/charts/kratos/values.yaml [hunk 3]
```diff
@@ -324,6 +324,11 @@ statefulSet:
   #    cpu: 100m
   #  memory: 128Mi
 
+  extraArgs: []
+  # -- Array of extra arguments to be passed down to the StatefulSet. Kubernetes args format is expected
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:328]:
```suggestion
  # -- Array of extra arguments to be passed down to the StatefulSet. Kubernetes args format is expected
  extraArgs: []
```

Comment (Demonsthere) [reply helm/charts/kratos/values.yaml:328]: The helm-docs app we use for docs generation requires the comment be above the variable :)

**Comment by Demonsthere** [comment]
I see, good catch with the `statefulset`, looks like a leftover that was missed when we were refactoring the chart. It is a breaking change, but for the sake of consistency I am in favour of merging it

**Review by Demonsthere — APPROVED** [review]

**Pull request merged**